controllers/storageconsumer/storageconsumer_controller.go

```diff
@@ -308,6 +316,60 @@ func (r *StorageConsumerReconciler) reconcileEnabledPhases() (reconcile.Result,
 
 	} else {
 		r.storageConsumer.Status.State = v1alpha1.StorageConsumerStateDeleting
+		_, hasForceDeleteAnnotation := r.storageConsumer.GetAnnotations()[util.ForceDeletionAnnotationKey]
+
+		consumerOwners := 0
+		for i := range consumerConfigMap.OwnerReferences {
+			if consumerConfigMap.OwnerReferences[i].Kind == "StorageConsumer" {
+				consumerOwners += 1
+			}
+		}
+
+		if consumerOwners <= 1 && hasForceDeleteAnnotation {
+			blockPools := &rookCephv1.CephBlockPoolList{}
+			if err := r.List(r.ctx, blockPools, client.InNamespace(r.namespace)); err != nil {
+				return reconcile.Result{}, err
+			}
+			consumerResources := util.WrapStorageConsumerResourceMap(consumerConfigMap.Data)
+			radosNamespaceName := consumerResources.GetRbdRadosNamespaceName()
+
+			patchPayload := fmt.Sprintf(`{"metadata":{"annotations":{"%s":"true"}}}`, util.RookForceDeletionAnnotationKey)
+			annotationPatch := client.RawPatch(types.MergePatchType, []byte(patchPayload))
+
+			for idx := range blockPools.Items {
+				bp := &blockPools.Items[idx]
+				rns := &rookCephv1.CephBlockPoolRadosNamespace{}
+				rns.Name = fmt.Sprintf("%s-%s", bp.Name, radosNamespaceName)
+				if radosNamespaceName == util.ImplicitRbdRadosNamespaceName {
+					rns.Name = fmt.Sprintf(
+						"%s-builtin-%s",
+						bp.Name,
+						util.ImplicitRbdRadosNamespaceName[1:len(util.ImplicitRbdRadosNamespaceName)-1],
```

Comment:
Why aren't we reading the name from the configmap. What is this `buildin` prefix doing here?

Reply:
This is the special case for the internal client, where the radosnamespacename is <implicit>
The CR name cannot have '<' '>' hence the name for the implicit raodsnamespace for each blockpool is blockpoolname-builtin-implicit

This is what we were doing while creating the radosnamespace https://github.com/red-hat-storage/ocs-operator/blob/main/controllers/storageconsumer/storageconsumer_controller.go#L454-L463

Reply:
This implementations defeats the purpose of comment you were referring to but I do agree that a change for this need to be provided in a different PR. Resolving